kyuubi-common/src/main/scala/org/apache/kyuubi/operation/log/OperationLog.scala

```diff
@@ -83,10 +83,10 @@ object OperationLog extends Logging {
 
 class OperationLog(path: Path) {
 
-  private val writer = Files.newBufferedWriter(path, StandardCharsets.UTF_8)
-  private val reader = Files.newBufferedReader(path, StandardCharsets.UTF_8)
+  private lazy val writer = Files.newBufferedWriter(path, StandardCharsets.UTF_8)
```

Comment:
The lazy creating saves open files resource, by may cause NPE if we set a high threshold for loggers.
Simply reverting the lazy change should resolve the issue, right?

Another solution I think we can introduce a flag `initialized` to indicate where we start writing logs to the file, `initialized` starts w/ false and should be updated to true after writing the first line to the log file. The reader should get an empty array instead of an exception if `initialized` is false.

thought? @ulysses-you @Yikf

Reply:
+1 for the initialized flag